content/en/docs/Architecture/cloud-init.md

```diff
@@ -41,32 +41,51 @@ stages:
+| _after-upgrade_            | This stage happens after upgrade has been done                                                                                                                                                                                                                                      |
+| _after-upgrade-chroot_     | This stage happens after the image has been upgraded                                                                                                                                                                                                                                |
 
 Note: Steps executed at the `chroot` stage are running inside the new OS as chroot, allowing to write persisting changes to the image, for example by downloading and installing additional software.
 
+{{% alert title="About after and before substages" color="info" %}}
+All the above stages are automatically run with a `before` and `after` suffix automatically, so users can always run something before of after the given
+stage with ease. So if a configuration is doing something in the `before-install` stage and you want to runs something before it, you could use the `before-install.before` to run it before the main stage runs.
+{{% /alert %}}
+
+
 In case you're using a standard image, with the Kairos provider, then these other stages are also available
 
-| **Stage**              | **Description**                                                                                                                                                                                                                                                                    |
-|------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| _provider-kairos.bootstrap.before.<role>_               | The provider fires this stage before starting to bootstrap K3S.                                                 |
-| _provider-kairos.bootstrap.after.<role>_               | The provider fires this stage after it finished bootstrapping K3S.                                                 |
+| **Stage**                                 | **Description**                                                    |
+|-------------------------------------------|--------------------------------------------------------------------|
+| _provider-kairos.bootstrap.before.<role>_ | The provider fires this stage before starting to bootstrap K3S.    |
+| _provider-kairos.bootstrap.after.<role>_  | The provider fires this stage after it finished bootstrapping K3S. |
+
+
+### Stages during kairos-agent operations in detail
+
+![Install Stages](https://github.com/user-attachments/assets/b050f247-990a-4395-9b45-334e04f84d45)
```

Comment:
<img width="370" alt="image" src="https://github.com/user-attachments/assets/7c640b49-018b-44d5-98f7-38396179feba">
does this mean that "Encrypt partitions" happens after "after-install" ? Maybe `after-install` should appear before `Encrypt partitions` box somehow? Something to make it more clear, I'm not a design expert for sure :D

Reply:
yes this is difficult to explain. I tried to convey that after encrypt aprtitions, the `after-install` stage runs but it doesnt seems like its clear enough.

@mauromorales you got the best design vision I think, any ideas how to make this clearer?

Reply:
thanks for thinking so highly of me haha

and I think the problem would be fixed if we remove those `before-install` and `after-install` and just have the stages with their names and `.pre_` and `.after_` or ideally `.before_` and `.after_` suffixes.

Since that is probably a hard change I think I would go with something like this for the time being:

- [mount partitions]
- [before-install]
- [deploy active image]
- [after-install-chroot]
- ...
- [encrypt partitions]
- [after-install]
- [reboot/power off]